### PR TITLE
446 - Change Type Of Work to About Your Organization

### LINKED ISSUE #446: Remove the "Type of work" question until we have a controlled list of values
Slack thread: https://gsa-tts.slack.com/archives/C03QM0JGSQG/p1677702708763699

**Acceptance Criteria**

- Remove the type of work question
- Replace with "About your organization..." question. [Updated text to replace "Type of work."](https://docs.google.com/document/d/1GTTeRvKfe3lA_w2U4X_CBk8Q7BpMe7W0m_eVZsTHzEE/edit#heading=h.3whwml4)

Related to https://github.com/cisagov/getgov/issues/447

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #446 

## Changes
1. Changing all the labeling from "Type Of Work" to "About Your Organization" + [new wording for About Your Organization](https://docs.google.com/document/d/1GTTeRvKfe3lA_w2U4X_CBk8Q7BpMe7W0m_eVZsTHzEE/edit#heading=h.3whwml4)
2. Migration for changing "Type Of Work" to "About Your Organization" in models
3. Remove any references to "more about organization"
4. "Type Of Work" wording in emails to "About Your Organization"

### **Question:**
In `src/registrar/forms/application_wizard.py`, I left a comment regarding an error message (see "TODO-446"). Would love help confirming if this wording is ok or if there is a more preferred wording let me know!

## Context for reviewers

We want to rename the "Type Of Work" section to "About Your Organization" and the accompany text with it -- [updated text linked here](https://docs.google.com/document/d/1GTTeRvKfe3lA_w2U4X_CBk8Q7BpMe7W0m_eVZsTHzEE/edit#heading=h.3whwml4)

## Setup

1. Log in locally or on staging 
2. Create a new domain 
3. For "Type of organization" click "Interstate"
4. For "Organization name and mailing address" fill in anything that suffices
5. "About your organization" will show up :)

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [x] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [x] Checked that the design translated visually
- [x] Checked behavior
- [x] Checked different states (empty, one, some, error)
- [x] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots
**Previous of what it looked like for "type of work":**
OLD_Typeofwork
**With the new PR changes:**
(Changes include the url path, form naming, and complete form information change
Screenshot 2023-09-08 at 1 45 06 PM

### CODE REVIEW

**Comment by zandercymatics** [comment]
Won't do a direct review yet as this is still a WIP, but your changes look good so far. This seems to be working for me locally

File: src/registrar/templates/application_about_your_organization.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{% extends 'application_form.html' %}
+{% load field_helpers %}
+
+{% block form_instructions %}
+  <p>[For special districts, interstate governments]</p>
```
Comment (zandercymatics) [inline src/registrar/templates/application_about_your_organization.html:5]: Would this be better as an h2 (like the other pages), without the square brackets? Ignore this though if this matches the design requirements

**Review by zandercymatics — CHANGES_REQUESTED** [review]
Great changes! I think this came together nicely. That said, I think I found a bug: If you change the type of organization while the application is in progress, some weird stuff happens:
image

If you just click on the 'About your Organization' tab, then navigate away, the page lets you submit it without any information in it, and it is shown as 'valid':
image
image
image

Steps to reproduce: either go to a started application or create one. For the purposes of this example, set it the type of org to Federal. Navigate past the point where 'About your Organization' would be on the menu (for example, Your contact information). Then, go back to 'Type of organization' and set it to 'Special District'.

Personally I am not entirely sure if this bug would fall under the scope of this ticket, I think its an issue with how that logic is implemented (edge case) - not anything specific to your additions. Maybe something we can create a ticket for?

That said, I'd ask @abroddrick or @rachidatecs as to their thoughts if this should be addressed here.

File: src/registrar/templates/application_about_your_organization.html (hunk 1 above)
Comment (Katherine-Osos) [reply src/registrar/templates/application_about_your_organization.html:5]: That line should be removed altogether. It was a content design note and shouldn't appear as text in the UI.

**Comment by therealslimhsiehdy** [comment]
> Great changes! I think this came together nicely. That said, I think I found a bug: If you change the type of organization while the application is in progress, some weird stuff happens: image
> 
> If you just click on the 'About your Organization' tab, then navigate away (clicking another menu item), the page lets you submit it without any information in it, and it is shown as 'valid': image image image
> 
> Steps to reproduce: either go to a started application or create one. For the purposes of this example, set it the type of org to Federal. Navigate past the point where 'About your Organization' would be on the menu (for example, Your contact information). Then, go back to 'Type of organization' and set it to 'Special District'.
> 
> Personally I am not entirely sure if this bug would fall under the scope of this ticket, I think its an issue with how that logic is implemented (edge case) - not anything specific to your additions. Maybe something we can create a ticket for?
> 
> That said, I'd ask @abroddrick or @rachidatecs as to their thoughts if this should be addressed here.



> Steps to reproduce: either go to a started application or create one. For the purposes of this example, set it the type of org to Federal. Navigate past the point where 'About your Organization' would be on the menu (for example, Your contact information). Then, go back to 'Type of organization' and set it to 'Special District'.
> 
> Personally I am not entirely sure if this bug would fall under the scope of this ticket, I think its an issue with how that logic is implemented (edge case) - not anything specific to your additions. Maybe something we can create a ticket for?
> 
> That said, I'd ask @abroddrick or @rachidatecs as to their thoughts if this should be addressed here.

Great catch! I reproduced this on a different branch on all browsers without these changes, and this seems to be a common theme. 

We do get an error if we don't fill out the text box (as we should bc it's required), but if you navigate elsewhere and then navigate back one is able to bypass filling it out.

TLDR Seems to be an existing issue --  @rachidatecs @abroddrick: would it be preferred to try to resolve the bug in this ticket, or create a bug ticket for this issue? Would love both your thoughts!

**Comment by therealslimhsiehdy** [comment]
[Slack discussion](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1694452430438639) for creating a new ticket for the bug since it's a bit out of scope and would require a bit of digging!

[Bug ticket ](https://github.com/cisagov/getgov/issues/1011)is now created.

**Review by zandercymatics — APPROVED** [review]
Great work on this ticket, thank you for making that other bug ticket as well!

**Comment by rachidatecs** [comment]
So we're doing away with more_organization_information?

**Comment by therealslimhsiehdy** [comment]
@rachidatecs Yes I believe so -- originally we had 2 text boxes and top one was for `type of work` and the bottom one was for `more_organization_info`, and since we don't have that second text box anymore we won't need to save any of that info :D

**Review by gabydisarli — APPROVED** [review]
One small edit to the error messaging language (discussed in huddle) then good to go!
